sw/device/tests/consecutive_irqs_test.c

```diff
@@ -25,27 +24,14 @@ static dif_uart_t uart0;
 
 // These flags are used in the test routine to verify that a corresponding
 // interrupt has elapsed, and has been serviced.
-static bool uart_rx_overflow_handled = false;
-static bool uart_tx_empty_handled = false;
-static bool uart_handled_more_than_once = false;
-
-static size_t polled_uart_sink_func(void *data, const char *buf, size_t len) {
-  for (int i = 0; i < len; ++i) {
-    if (dif_uart_byte_send_polled(&uart0, (uint8_t)buf[i]) != kDifUartOk) {
-      abort();
-    }
-  }
-  return len;
-}
-
-static const buffer_sink_t kPolledUartSink = {
-    .data = NULL, .sink = &polled_uart_sink_func,
-};
-
-#define LOG_FATAL_AND_ABORT(...) \
-  do {                           \
-    LOG_FATAL(__VA_ARGS__);      \
-    abort();                     \
+static volatile bool uart_rx_overflow_handled;
+static volatile bool uart_tx_empty_handled;
+static volatile bool uart_handled_more_than_once;
```

Comment:
Why are these declared volatile? `volatile` is forbidden in preference to `mmio_region`.

Reply:
NM I realized why you need this, bu you should add a comment noting the reason (I initially thought it was for synchronization, which is an easy mistake to make with `volatile`.)